payments/schema/types.py

```diff
@@ -138,3 +149,90 @@ def resolve_product_type(self, info):
             if self.service.is_fixed_service()
             else AdditionalProductType.OPTIONAL_SERVICE
         )
+
+
+class ProductUnion(graphene.Union):
+    class Meta:
+        types = (BerthProductNode, WinterStorageProductNode)
+
+
+class LeaseUnion(graphene.Union):
+    class Meta:
+        # TODO: Add WinterStorageLeaseNode
+        types = (BerthLeaseNode,)
+
+
+class OrderNode(DjangoObjectType):
+    customer = graphene.Field("customers.schema.ProfileNode", required=True)
+    product = graphene.Field(ProductUnion)
+    lease = graphene.Field(LeaseUnion)
+    status = OrderStatusEnum(required=True)
+    comment = graphene.String()
+    price = graphene.Decimal(required=True)
+    tax_percentage = graphene.Decimal(required=True)
+    pretax_price = graphene.Decimal(required=True)
+    total_price = graphene.Decimal(required=True)
+    total_pretax_price = graphene.Decimal(required=True)
+    total_tax_percentage = graphene.Decimal(required=True)
+    due_date = graphene.Date(required=True)
+    order_lines = DjangoConnectionField("payments.schema.OrderLineNode", required=True)
+    log_entries = DjangoConnectionField(
+        "payments.schema.OrderLogEntryNode", required=True
+    )
+
+    class Meta:
+        model = Order
+        interfaces = (graphene.relay.Node,)
+        connection_class = CountConnection
+        exclude = (
+            "_product_content_type",
+            "_product_object_id",
+            "_lease_content_type",
+            "_lease_object_id",
+        )
+
+    @classmethod
+    @view_permission_required(Order)
+    def get_queryset(cls, queryset, info):
+        return super().get_queryset(queryset, info)
+
+    def resolve_order_lines(self, info, **kwargs):
+        return OrderLine.objects.filter(order=self)
+
+    def resolve_log_entries(self, info, **kwargs):
+        return OrderLogEntry.objects.filter(order=self)
```

Comment:
do you really need these resolvers? 🤔 isn't graphene `magic` enough to resolve them itself?

Reply:
Good catch! Smart smart, I didn't know if the filtering was gonna work, but it did. 👍

Reply:
it does not really filter, it just execute `self.<field_name>.all()` which is == to `self.order_lines.all()` and `self.log_entries.all()`. so as long as the field_name == related_name on the FK then it should be _magic_ 🧙‍♂️

Reply:
Aaaahh, alright, makes sense. 👍